### ISSUE TITLE
CA2000 on using declaration with object initializer

### ISSUE THREAD

**Issue by sectax** [dotnet/roslyn-analyzers#2703]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3 (Latest)

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Repro steps
1. Create a library using netstandard2.0 and `<LangVersion>preview</LangVersion>`
2. Insert following snippet code

```cs
public static void RaisesWarning()
{
    using var aesEncryption = new RijndaelManaged
    {
        KeySize = 128
    };
}

public static void DoesNotRaiseWarning()
{
    using (var aesEncryption = new RijndaelManaged
    {
        KeySize = 128
    }) {}
}
```

#### Expected behavior
Either both methods generate CA2000, or neither.

#### Actual behavior
Only the method with a using declaration generates a CA2000 warning.

**Comment by mavasani** [comment]
This is due to missing IOperation support for using declarations in Roslyn: https://github.com/dotnet/roslyn/issues/32100

**Comment by mavasani** [comment]
I will add a workaround to the analyzer for this case.
